docs/signatory-architecture.md

```diff
@@ -0,0 +1,295 @@
+
+Using Signatory, users can securely store their secret keys and control which operations can be signed, reducing the risk of losing or having their keys stolen. Signatory also allows users to sign transactions on hardware not connected to the internet, providing an additional layer of security.
+
+[C4 architecture diagrams](https://c4model.com/) are visual modelling languages used to communicate software system architectures. They are designed to be simple, easy to understand and provide a high-level view of a system's components and how they interact. C4 diagrams are typically organized into a hierarchy of increasing detail, with each view building on the previous one.
+
+[Domain-driven design](https://martinfowler.com/tags/domain%20driven%20design.html) ([DDD](https://learn.microsoft.com/en-us/archive/msdn-magazine/2009/february/best-practice-an-introduction-to-domain-driven-design)) is a software development approach emphasizing the importance of understanding the business domain to create effective software solutions. C4 diagrams can help developers understand the domain of the system they are building and design software that aligns with the domain concepts and principles. By using C4 diagrams to visualize the system's architecture, developers can better understand how the system fits into the larger business context and how the different components of the system interact with one another.
+
+The C4 system architectural model is a set of diagrams used to represent the architecture of a software system. There are four levels in the C4 model, each providing an increasing level of detail about the software system.
+
+1. System Context Diagram: This is the highest level of the C4 model and provides a high-level view of the software system and its environment. It shows the system as a box in the center, surrounded by its users and other systems it interacts with. The diagram also shows the relationships between the system and its environment, such as data flow and requests.
+2. Container Diagram: The container diagram shows the software system as a set of containers, each containing its own set of components. It shows how the different containers interact with each other to form a larger system. This level of the C4 model provides more detail about the software system's components and their interactions.
+3. Component Diagram: The component diagram provides a more detailed view of the software system's components and their interaction. It shows the internal structure of each container and the relationships between the components. This level of the C4 model provides a detailed view of the software system's components and their interactions.
+4. Code Diagram: This is the lowest level of the C4 model, and it provides a detailed view of the code structure of each component. It shows the classes, functions, and other code artifacts that make up each component.
```

Comment:
Do we need to describe what C4 diagrams are in points 1 through 4 on the Signatory website?  maybe a link to the C4 website is best

Reply:
yes, I think it shows clearly what we are discussing and what we want users to understand and provides context they can check back on easily.